data_linter/validation.py

```diff
@@ -156,31 +170,198 @@ def match_files_in_land_to_config(config: dict) -> dict:
+            bucket, key = s3_path_to_bucket_key(file_name)
+            obj = s3_client.get_object(Bucket=bucket, Key=key)
+            file_size = obj.get("ContentLength")
+            file_list[i]["file-size"] = file_size
+            acum_file_size += file_size
+
+        target_bin_size = acum_file_size / max_bin_count
+
+        # sort them in descending order
+        file_list.sort(key=lambda x: -x["file-size"])
+
+        bins = [None for i in range(max_bin_count)]
+
+        offset = 0
+        for i in range(max_bin_count):
+            curr_bin = []
+            curr_bin_size = 0
+            has_been_binned = False
+            for j in range(offset, len(file_list)):
+                if len(curr_bin) == 0:
+                    curr_bin.append(file_list[j])
+                    curr_bin_size += file_list[j]["file-size"]
+                    offset += 1
+                else:
+                    if curr_bin_size <= target_bin_size:
+                        curr_bin.append(file_list[j])
+                        curr_bin_size += file_list[j]["file-size"]
+                        offset += 1
+                    else:
+                        bins[i] = curr_bin
+                        has_been_binned = True
+                        break
+            if not has_been_binned:
+                bins[i] = curr_bin
+
+        bins[i] = curr_bin
+        bins = [i for i in bins if i != []]
+
+        # create the configs for the given bins
+        for i, packed_bin in enumerate(bins):
+            config_n = deepcopy(config)
+            config_n.pop("tables")
+            config_n["tables"] = {}
+
+            for table in packed_bin:
+                curr_table_name = table.pop("table-name")
+
+                if config_n["tables"].get(curr_table_name):
+                    # it exists, so just add to matched files
+                    config_n["tables"][curr_table_name]["matched_files"].append(
+                        table["file-name"]
+                    )
+                else:
+                    # it doesn't exist, do a full copy of all attributes
+                    mfile = table.pop("file-name")
+                    table.pop("file-size")
+                    config_n["tables"][curr_table_name] = deepcopy(table)
+                    config_n["tables"][curr_table_name]["matched_files"] = []
+                    config_n["tables"][curr_table_name]["matched_files"].append(mfile)
+
+            # upload the config to temp storage, into config
+            with tempfile.NamedTemporaryFile(
+                suffix=".yml", prefix="config_"
+            ) as tmp_file:
+
+                with open(tmp_file.name, "w") as yaml_out:
+                    yaml.dump(config_n, yaml_out, default_flow_style=False)
+
+                tmp_file_name = tmp_file.name.split("/")[-1]
+                local_file_to_s3(tmp_file.name, f"{s3_temp_path}/{i}/{tmp_file_name}")
+
+    else:
+        pass
+        # do local equiv maybe lol
```

Comment:
Should raise error here

Reply:
do you think this should be made to work locally like `collect_all_status` instead?

Reply:
Changed to raise error instead